compiler/ghc/default.nix

```diff
@@ -259,7 +295,7 @@ stdenv.mkDerivation (rec {
     '' + lib.optionalString (ghc-version-date != null) ''
         substituteInPlace configure --replace 'RELEASE=YES' 'RELEASE=NO'
         echo '${ghc-version-date}' > VERSION_DATE
-    '' + lib.optionalString (builtins.compareVersions ghc-version "9.2.3" >= 0) ''
+    '' + lib.optionalString (builtins.compareVersions ghc-version "9.2.3" >= 0 && builtins.compareVersions ghc-version "9.4" < 0) ''
```

Comment:
We don't need boot for 9.4+? Is that true? I think the `boot` script has a `--hadrian` flag, to only boot the Hadrian related stuff?

Reply:
Turns out this is only needed for ghc 9.2.3 because the tarball was unbooted.   Conditional improved and comment added.